src/Microsoft.Health.SqlServer/Features/Schema/SchemaInitializer.cs

```diff
@@ -167,22 +163,24 @@ public static async Task<bool> DoesDatabaseExistAsync(SqlConnection sqlConnectio
             return false;
         }
 
+        [SuppressMessage("Security", "CA2100:Review SQL queries for security vulnerabilities", Justification = "Database name is validated before use.")]
         public static async Task<bool> CreateDatabaseAsync(SqlConnection connection, string databaseName, CancellationToken cancellationToken)
         {
-            using (var canCreateDatabaseCommand = new SqlCommand("SELECT count(*) FROM fn_my_permissions (NULL, 'DATABASE') WHERE permission_name = 'CREATE DATABASE'", connection))
+            if (!Identifier.IsValidDatabase(databaseName))
```

Comment:
This was the rare exception in the code where we cannot use a parameterized query, so to avoid any malicious `databaseName` values I validate it via a new helper method.